pennylane/gradients/hamiltonian_grad.py

```diff
@@ -24,7 +24,7 @@ def hamiltonian_grad(tape, idx):
         tape (qml.tape.QuantumTape): tape with a single Hamiltonian expectation as measurement
         idx (int): index of parameter that we differentiate with respect to
     """
-    op, p_idx = tape.get_operation(idx)
+    op, _, p_idx = tape._get_operation(idx)
```

Comment:
Wouldn't it be more consistent to use the new keyword argument here, and in particular its default? :thinking:

Reply:
Same for the changes in `parameter_shift.py`

Reply:
+1.  We don't want to be relying on private methods outside of the class itself.